src/ensembl/src/content/app/entity-viewer/state/gene-view/proteins/geneViewProteinsSlice.ts

```diff
@@ -107,7 +107,11 @@ const proteinsSlice = createSlice({
     ) {
       const { activeGenomeId, activeObjectId, expandedIds } = action.payload;
       if (!state[activeGenomeId]) {
-        state[activeGenomeId] = { [activeObjectId]: defaultStatePerGene };
+        state[activeGenomeId] = {
+          [activeObjectId]: { ...defaultStatePerGene }
+        };
+      } else if (!state[activeGenomeId][activeObjectId]) {
+        state[activeGenomeId][activeObjectId] = { ...defaultStatePerGene };
```

Comment:
I don't think we need the spread, either here or on line 111.

Reply:
I get the following error when I don't use the spread:

`TypeError: Cannot assign to read only property 'expandedTranscriptIds' of object '#<Object>'
    at updateExpandedProteins (geneViewProteinsSlice.ts?773c:116)`

Am I missing something?

Reply:
Ah, immer! And we are not returning the state, but modifying it further on line 116. In this case, both you and immer are right 😄